perfkitbenchmarker/linux_benchmarks/terasort_benchmark.py

```diff
@@ -0,0 +1,136 @@
+
+This benchmark takes runs the Apache Terasort benchmark on an
+Apache Spark cluster. The Apache Spark cluster can be one
+supplied by a cloud provider, such as Google's Dataproc.
+
+It records how long each phase (generate, sort, validate) takes to run.
+For each phase, it reports the wall clock time, but this number should
+be used with caution, as it some platforms (such as AWS's EMR) use polling
+to determine when the job is done, so the wall time is inflated
+Furthermore, if the standard output of the job is retrieved, AWS EMR's
+time is again inflated because it takes extra time to get the output.
+
+If available, it will also report a pending time (the time between when the
+job was received by the platform and when it ran), and a runtime, which is
+the time the job took to run, as reported by the underlying cluster.
+
+For more on Apache Spark, see: http://spark.apache.org/
+"""
+
+import datetime
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import sample
+from perfkitbenchmarker import spark_service
+from perfkitbenchmarker import flags
+
+
+
+BENCHMARK_NAME = 'terasort_benchmark'
+BENCHMARK_CONFIG = """
+terasort_benchmark:
+  description: Run the Apache hadoop terasort benchmark on a cluster.
+  spark_service:
+    service_type: managed
+    num_workers: 4
+"""
+
+# This points to a file on the spark cluster.
+TERASORT_JARFILE = ('file:///usr/lib/hadoop-mapreduce/'
+                    'hadoop-mapreduce-examples.jar')
+TERAGEN_CLASSNAME = 'org.apache.hadoop.examples.terasort.TeraGen'
+TERASORT_CLASSNAME = 'org.apache.hadoop.examples.terasort.TeraSort'
+TERAVALIDATE_CLASSNAME = 'org.apache.hadoop.examples.terasort.TeraValidate'
+
+flags.DEFINE_integer('terasort_dataset_size', 10000,
+                     'Data set size to generate')
+flags.DEFINE_string('terasort_unsorted_dir', 'tera_gen_data', 'Location of '
+                    'the unsorted data. TeraGen writes here, and TeraSort '
+                    'reads from here.')
+
+flags.DEFINE_string('terasort_sorted_dir', 'tera_sort_dir', 'Location for the '
+                    'sorted data. TeraSort writes to here, TeraValidate reads '
+                    'from here.')
+flags.DEFINE_string('terasort_validate_dir', 'tera_validate_dir', 'Output of '
+                    'the TeraValidate command')
+
+FLAGS = flags.FLAGS
+
+
+def GetConfig(user_config):
+  return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+
+
+def Prepare(benchmark_spec):
+  pass
+
+
+def Run(benchmark_spec):
+  """Executes the given jar on the specified Spark cluster.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  spark_cluster = benchmark_spec.spark_service
+  start = datetime.datetime.now()
+
+  results = []
+  metadata = spark_cluster.GetMetadata()
+  gen_args = [str(FLAGS.terasort_dataset_size), FLAGS.terasort_unsorted_dir]
```

Comment:
This is more of a style thing but I tend to put lists that should be immutable into tuples, in this case 
``` gen_args = str(FLAGS.terasort_dataset_size), FLAGS.terasort_unsorted_dir```
so someone doesn't pop or append a value.